File: be/src/main/java/sg/flow/entities/utils/CardType.java
package sg.flow.entities.utils;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum CardType {
    CREDIT("CREDIT"),
    DEBIT("DEBIT"),
    PREPAID("PREPAID"),
    CORPORATE("CORPORATE"),
    ATM("ATM"),
    OTHERS("OTHERS");

    private final String value;

    CardType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return value;
    }

    @JsonCreator
    public static CardType fromValue(String value) {
        for (CardType type : CardType.values()) {
            if (type.value.equalsIgnoreCase(value)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown enum type " + value);
    }
}

File: be/src/main/kotlin/sg/flow/entities/utils/CardType.kt
package sg.flow.entities.utils

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class CardType(private val value: String) {
    CREDIT("CREDIT"),
    DEBIT("DEBIT"),
    PREPAID("PREPAID"),
    CORPORATE("CORPORATE"),
    ATM("ATM"),
    OTHERS("OTHERS");

    @get:JsonValue
    val getValue: String
        get() = value

    companion object {
        @JvmStatic
        @JsonCreator
        fun fromValue(value: String): CardType {
            return values().firstOrNull { it.value.equals(value, ignoreCase = true) }
                    ?: throw IllegalArgumentException("Unknown enum type $value")
        }
    }
}
